lib/Document/Renderer/DocumentRenderer.php

```diff
@@ -87,7 +89,23 @@ public function render(Document\PageSnippet $document, array $attributes = [], a
         try {
             $request = $this->requestHelper->getCurrentRequest();
         } catch (\Exception $e) {
-            $request = $this->requestHelper->createRequestWithContext();
+
+            $host = 'localhost';
+            if( $site = Frontend::getSiteForDocument($document) ){
+                $host = $site->getMainDomain();
+            }elseif( $systemMainDomain = Tool::getHostUrl() ){
+                $host = $systemMainDomain;
+            }
+
+            $request = $this->requestHelper->createRequestWithContext(host: $host);
+        }
+
+        if( $attributes['pimcore_static_page_generator']){
+            $config = \Pimcore::getContainer()->getParameter('pimcore.config');
+            $headers = $config['static_page_generator']['headers'];
```

Comment:
```suggestion
            $headers = \Pimcore\Config::getSystemConfiguration('documents')['static_page_generator']['headers'];
```
Based on config node changes